osv2/mget_test.go

```diff
@@ -0,0 +1,312 @@
+			}
+
+			require.JSONEq(t, string(tt.want), string(got))
+		})
+	}
+}
+
+func Test_MGetResult_ToModel(t *testing.T) {
+	type fields struct {
+		Index       string
+		ID          string
+		Version     int
+		SeqNo       int
+		PrimaryTerm int
+		Found       bool
+		Source      json.RawMessage
+		Error       error
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   opensearchtools.MGetResult
+	}{
+		{
+			name: "Non-error result",
+			fields: fields{
+				Index:       testIndex1,
+				ID:          testID1,
+				Version:     42,
+				SeqNo:       99,
+				PrimaryTerm: 10,
+				Found:       true,
+				Source:      []byte(`{"name": "bob", "age": 42}`),
+				Error:       nil,
+			},
+			want: opensearchtools.MGetResult{
+				Index:       testIndex1,
+				ID:          testID1,
+				Version:     42,
+				SeqNo:       99,
+				PrimaryTerm: 10,
+				Found:       true,
+				Source:      []byte(`{"name": "bob", "age": 42}`),
+				Error:       nil,
+			},
+		},
+		{
+			name: "Error result",
+			fields: fields{
+				Error: fmt.Errorf("some OpenSearch error"),
+			},
+			want: opensearchtools.MGetResult{
+				Error: fmt.Errorf("some OpenSearch error"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			marshalableResult := &MGetResult{
+				Index:       tt.fields.Index,
+				ID:          tt.fields.ID,
+				Version:     tt.fields.Version,
+				SeqNo:       tt.fields.SeqNo,
+				PrimaryTerm: tt.fields.PrimaryTerm,
+				Found:       tt.fields.Found,
+				Source:      tt.fields.Source,
+				Error:       tt.fields.Error,
+			}
+			got := marshalableResult.ToModel()
+			require.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_MGetResponse_ToModel(t *testing.T) {
+	testHeaders := http.Header{}
+	testHeaders.Add("x-foo", "bar")
+
+	type fields struct {
+		MarshlableResponse MGetResponse
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   *opensearchtools.MGetResponse
+	}{
+		{
+			name: "Multiple docs returned",
+			fields: fields{
+				MarshlableResponse: MGetResponse{
+					StatusCode: 200,
+					Header:     testHeaders,
+					Docs: []MGetResult{
+						{
+							Index:       testIndex1,
+							ID:          testID1,
+							Version:     42,
+							SeqNo:       99,
+							PrimaryTerm: 10,
+							Found:       true,
+							Source:      []byte(`{"name": "bob", "age": 42}`),
+							Error:       nil,
+						},
+						{
+							Index:       testIndex2,
+							ID:          testID2,
+							Version:     1,
+							SeqNo:       2,
+							PrimaryTerm: 2,
+							Found:       true,
+							Source:      []byte(`{"deviceName": "abc123", "os": "windows"}`),
+							Error:       nil,
+						},
+						{
+							Index:       testIndex2,
+							ID:          testID2,
+							Version:     10,
+							SeqNo:       220,
+							PrimaryTerm: 30,
+							Found:       false,
+							Source:      []byte{},
+							Error:       nil,
+						},
+					},
+				},
+			},
+			want: &opensearchtools.MGetResponse{
```

Comment:
let's add a test for an error case where no results are returned.

Reply:
done